config/dynamic.c

```diff
@@ -16,11 +16,467 @@
+	}
+
+	return 0;
+}
+
+static int
+process_client_message(int conn_fd,
+	const char *msg, int msg_len, struct dynamic_config *dy_conf)
+{
+	/*
+	 * TODO Implement the functionalities to process the clients' request.
+	 *
+	 * Gatekeeper: Adding, listing, and removing BP components;
+	 *
+	 * Gatekeeper: Adding, listing, removing IP ranges
+	 * handled by the global LPM table in GKs;
+	 *
+	 * Gatekeeper and Grantor: Listing the ARP and ND table.
+	 * This is important for network diagnosis.
+	 *
+	 * Grantor: Updating the policy GTs run.
+	 */
+
+	int ret;
+	int reply_len;
+	char reply_msg[LUA_MSG_MAX_LEN + 1];
+	memset(reply_msg, 0, sizeof(reply_msg));
+
+	RTE_LOG(NOTICE, GATEKEEPER,
+		"dyn_cfg: Received message (len = %d) - \n %s\n",
+		msg_len, msg);
+
+	ret = luaL_dostring(dy_conf->lua_state, msg);
+	if (ret != 0) {
+		RTE_LOG(ERR, GATEKEEPER,
+			"dyn_cfg: %s!\n", lua_tostring(dy_conf->lua_state, -1));
+		return -1;
+	}
+
+	/* Function to be called. */
+	lua_getglobal(dy_conf->lua_state, "dy_cmd_proc");
+	lua_pushlightuserdata(dy_conf->lua_state, reply_msg);
+	ret = lua_pcall(dy_conf->lua_state, 1, 1, 0);
+	if (ret != 0) {
+		RTE_LOG(ERR, GATEKEEPER,
+			"dyn_cfg: %s!\n", lua_tostring(dy_conf->lua_state, -1));
+		return -1;
+	}
+
+	ret = luaL_checkinteger(dy_conf->lua_state, -1);
+	if (ret < 0) {
+		RTE_LOG(ERR, GATEKEEPER,
+			"dyn_cfg: dy_cmd_proc() return value is %d!\n",
+			ret);
+		reply_client_message(conn_fd,
+			CLIENT_PROC_ERROR, strlen(CLIENT_PROC_ERROR));
+		return -1;
+	}
+
+	reply_len = strlen(reply_msg);
+	if (reply_len > LUA_MSG_MAX_LEN) {
+		RTE_LOG(ERR, MALLOC,
+			"dyn_cfg: The reply message length (%d) exceeds the limit!\n",
+			reply_len);
+		reply_client_message(conn_fd,
+			CLIENT_PROC_ERROR, strlen(CLIENT_PROC_ERROR));
+		return -1;
+	}
+
+	return reply_client_message(conn_fd, reply_msg, reply_len);
+}
+
+/*
+ * Return values:
+ * 	-1: Error happens.
+ * 	0: Command successfully processed, may need to process further commands.
+ * 	1: Connection closed by the client.
+ */
+static int
+process_single_cmd(int conn_fd, char *msg_buff,
+	const char *client_ip, struct dynamic_config *dy_conf)
+{
+	int recv_size;
+	int tot_size = 0;
+	uint16_t msg_len = 0;
+
+	/*
+	 * The protocol should be rather simple: two-byte,
+	 * unsigned integer in network order signal the size,
+	 * in bytes, of the message that follows that
+	 * first two bytes.
+	 */
+	while ((recv_size = recv(conn_fd,
+			(char *)&msg_len + tot_size, 1, 0)) > 0) {
+		tot_size += recv_size;
+		if (tot_size >= 2)
+			break;
+	}
+
+	/* The connection with the client is closed. */
+	if (recv_size == 0) {
+		RTE_LOG(NOTICE, GATEKEEPER,
+			"dyn_cfg: Client %s disconnected!\n",
+			client_ip);
+		return 1;
+	}
+
+	msg_len = ntohs(msg_len);
+	if (msg_len > LUA_MSG_MAX_LEN) {
+		RTE_LOG(ERR, MALLOC,
+			"dyn_cfg: The message length (%d) exceeds the limit!\n",
+			msg_len);
+		return -1;
+	}
+
+	tot_size = 0;
+	memset(msg_buff, 0, sizeof(char) * (LUA_MSG_MAX_LEN + 1));
```

Comment:
Is the `memset()` doing anything useful since `recv()` is called on the buffer `msg_buff` on the next line?